test/client_reset_test.dart

```diff
@@ -454,6 +456,11 @@ Future<void> main([List<String>? args]) async {
     expect(manualResetFallbackOccurred, 1);
     expect(onAfterResetOccurred, 1);
     expect(onBeforeResetOccurred, 1);
+
+    expect(clientResetErrorOnManualFallback.category, SyncErrorCategory.client);
+    expect(clientResetErrorOnManualFallback.isAutomaticClientReset, isTrue);
+    print(clientResetErrorOnManualFallback.code);
+    expect(clientResetErrorOnManualFallback.code, SyncSessionErrorCode.unknown);
```

Comment:
When a client reset fails and a manual fallback occurs, then `isAutomaticClientReset` is true, but `SyncSessionErrorCode` is `unknown` because the error occurred on the client side.

Reply:
This seems like a good candidate for a code comment